apps/api/src/modules/fixtures/fixtures.controller.ts

```diff
@@ -0,0 +1,198 @@
+import { Request, Response, NextFunction } from "express";
+import { pool } from "../../database";
+import * as fixturesService from "./fixtures.service";
+import {
+  createFixtureSchema,
+  fixtureIdParamSchema,
+  squadIdParamSchema,
+  setAvailabilitySchema,
+} from "./fixtures.validators";
+
+type Role = "COACH" | "PLAYER";
+type ReqUser = { id: number; email: string; role: Role };
+
+export async function createFixture(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const user = (req as any).user as ReqUser | undefined;
+    if (!user) {
+      return next(Object.assign(new Error("Unauthorized"), { status: 401 }));
+    }
+    if (user.role !== "COACH") {
+      return next(
+        Object.assign(new Error("Forbidden: Coach only"), { status: 403 })
+      );
+    }
+
+    const body = createFixtureSchema.parse(req.body);
+
+    const squadId = Number(req.body?.squadId);
+    if (!squadId || Number.isNaN(squadId)) {
+      return next(
+        Object.assign(new Error("squadId is required and must be a number"), {
+          status: 400,
+        })
+      );
+    }
+
+    const payload: {
+      squadId: number;
+      opponent: string;
+      kickoffAt: string;
+      actingCoachId: number;
+      location?: string;
+      notes?: string;
+    } = {
+      squadId,
+      opponent: body.opponent,
+      kickoffAt: body.kickoffAt,
+      actingCoachId: user.id,
+    };
+
+    if (body.location) {
+      payload.location = body.location;
+    }
+    if (body.notes) {
+      payload.notes = body.notes;
+    }
+
+    const fixture = await fixturesService.createFixture(payload);
+
+    res.status(201).json(fixture);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function getFixture(
```

Comment:
This controller is handling too much (auth check, DB queries, access rules, response). Controllers should just parse the request and send the response. The squad ownership/membership checks and queries belong in the service. That way the controller stays clean and reusable.

ex this should look something like:

```
export async function getFixture(req, res, next) {
  try {
    const user = (req as any).user
    const { id } = fixtureIdParamSchema.parse(req.params)

    const fixture = await fixturesService.getFixtureWithAccessCheck(id, user)

    res.json(fixture)
  } catch (err) {
    next(err)
  }
}
```